File: 258_Server/258_Server.py
def solution(players, m, k):
    answer = 0
    server = []
    
    for player in players:
        server = [val - 1 for val in server]
        server = [val for val in server if val > 0]

        available_players = len(server) * m + m - 1
        if player > available_players:
            temp = available_players
            while temp < player:
                temp += m
                server.append(k)
                answer += 1
    return answer

File: 258_Server/258_Server.js
function solution(players, m, k) {
  let answer = 0;
  let server = [];

  for (const player of players) {
    server.forEach((val, idx) => server[idx]--);
    server = server.filter((val) => val > 0);

    const available_players = server.length * m + m - 1;
    if (player > available_players) {
      let temp = available_players;
      while (temp < player) {
        server.push(k);
        temp += m;
        answer++;
      }
    }
  }
  return answer;
}
